continuous-integration/starter-kits/README.md

```diff
@@ -1,6 +1,177 @@
+and customization on local workstations.)
+* Administrative access to a **Github** repository that is the build source.
+* **Java** and **Maven** installed locally to test build conversions.
+* **Docker Desktop** to test container executions
+* Access to a **Jenkins** server, such as the enterprise CAE Jenkins at JPL.
+* Access to an artifact repository, such as the enterprise CAE Artifactory at JPL.
+
+For both examples described below, see the raw source which you can copy/paste into your own 
+project to configure a working build system.
+
+There are two configured examples: _(Maven setups supporting application designs)_
+1. `simple`: A monolithic compile of a single application that produces a single build artifact.
+2. `multimodule`: An multi-step build that compiles several modules and uses them as dependencies 
+in subsequent steps -- each step may produce its own artifacts, e.g. `JAR`s, `WAR`s or 
+other packaged build output. 
+
+### Requirements
+CI creates an ecosystem between products such that servers in compile, release and publishing 
+phases handoff processing between steps. All implementations are predicated on the following 
+configurations:
+1. Setup an account on a Jenkins server to host the build, such as the enterprise CAE Jenkins server 
+at JPL or another shared Jenkins host. 
+2. [Setup Webhook notifications](https://github.blog/2014-02-11-webhooks-level-up/) pointing to the 
+Jenkins instance with the URL of the server from the previous step that will host the build.
+3. Setup an account on an Artifactory server to host built binaries, such as the enterprise CAE 
+Artifactory server at JPL or another similar repository.
+4. Setup authorization credentials to interconnect services.
+   1. Setup [Github SSH Key Access](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh) 
+   and record the key for later use.
+   2. Create an [API Key in Artifactory](https://www.jfrog.com/confluence/display/JFROG/User+Profile#UserProfile-APIKey) 
+   and record it for later use.
+   3. Create a [Maven Settings File](https://www.jfrog.com/confluence/display/JFROG/Maven+Repository) 
+   using the previously generated Artifactory API Key
+5. [Configure Jenkins Global Credentials](https://www.jenkins.io/doc/book/using/using-credentials/#configuring-credentials) 
+by creating an entry for each of the Github SSH key and Maven Settings File previously generated.
+
+#### Base configs and alternatives
+These examples rely on certain assumptions about underlying tools. For instance, a 
+[Dockerfile](https://github.com/nasa-ammos/slim/blob/main/continuous-integration/starter-kits/compile-release-publish/java-maven/simple/.ci/Dockerfile.buildDeploy) 
+referenced in one example presumes that Jenkins uses a certain UID and GID set up for a particular 
+Jenkins server. These values may be changed based on any chosen target server. 
```

Comment:
"... or may not be necessary at all"
(just a suggested continuation of that last sentence)